src/dlload.c

```diff
@@ -435,6 +435,13 @@ JL_DLLEXPORT int jl_dlsym(void *handle, const char *symbol, void ** value, int t
 // Look for symbols in internal libraries
 JL_DLLEXPORT const char *jl_dlfind(const char *f_name)
 {
+#ifdef _OS_FREEBSD_
+    // This is a workaround for FreeBSD <= 13.2 which do not have
+    // https://cgit.freebsd.org/src/commit/?id=21a52f99440c9bec7679f3b0c5c9d888901c3694
+    // (See https://github.com/JuliaLang/julia/issues/50846)
+    if (strcmp(f_name, "dl_iterate_phdr") == 0)
```

Comment:
This seems reasonable to me as a stopgap solution but I wonder if there's a way we can more directly query whether `dl_iterate_phdr` resolved to libc does the right thing so that we don't have to perform this check on every call.

Reply:
I expect this check is generally a lot cheaper than the `dlsym`s below, but thankfully the standard function/library pointer caching for `ccall` should mean we only do this once per call-site